File: src/main/java/fi/finlit/edith/ui/services/PersonDao.java
package fi.finlit.edith.ui.services;

import java.util.Collection;

import org.apache.tapestry5.hibernate.annotations.CommitAfter;

import fi.finlit.edith.sql.domain.Person;

public interface PersonDao extends Dao<Person, Long> {

    Collection<Person> findByStartOfFirstAndLastName(String partial, int limit);

    @CommitAfter
    void remove(Long personId);

    @CommitAfter
    void save(Person person);

}

File: src/main/scala/fi/finlit/edith/ui/services/PersonDao.scala
package fi.finlit.edith.ui.services

import java.util.Collection
import org.apache.tapestry5.hibernate.annotations.CommitAfter
import fi.finlit.edith.sql.domain.Person
//remove if not needed
import scala.collection.JavaConversions._

trait PersonDao extends Dao[Person, Long] {

  def findByStartOfFirstAndLastName(partial: String, limit: Int): Collection[Person]

  @CommitAfter
  def remove(personId: java.lang.Long): Unit

  @CommitAfter
  def save(person: Person): Unit
}
